### ISSUE TITLE
deleting within a context manager does not persist..

### ISSUE THREAD

**Issue by vvasuki** [cloudant/python-cloudant#317]
With a couchdb database db, when I do:
```
    from cloudant.document import Document
    with Document(db, doc_id) as db_doc:
      db_doc.delete()
      logging.debug(db[doc_id].exists())
    logging.debug(db[doc_id].exists())
```
the first `logging.debug(db[doc_id].exists())` prints False, while the second prints True. This is rather strange, and points to a bug either in the context manager or the delete documentation.

**Comment by smithsz** [comment]
I would expect this behaviour. The `Document` context manager allows clients to make multiple updates to a document and only save to the server _once_ upon exiting. For example:

```python
with Document(db, doc_id) as doc:
    for user in users:
        doc['users'].append(user)
```

The `Document.delete()` method makes a server request to delete that document. It also calls `.clear()` on the local `Document` dictionary.

In your example above you are fetching the document from the server upon entering the context manager, deleting it from the server and then saving a new, empty document upon exiting the context manager.

If you only want to delete a document then you can simply do this:

```python
db[doc_id].delete()
```
_tl;dr Don't use the `Document.delete()` method inside the `Document` context manager._

Hope this helps. Let me know if you have further questions.

**Comment by ricellis** [comment]
The [docs](http://python-cloudant.readthedocs.io/en/stable/getting_started.html#context-managers) describe the entry/exit fetch/save behaviour, but I think a warning about this deletion quirk would be valuable.
